### PR TITLE
Add Stack Implementation in C

### PR DESCRIPTION
**Pull request by hemanth-kotagiri**

### CODE REVIEW

File: ds/linked_list/src/ll.c [hunk 1]
```diff
@@ -1,14 +1,60 @@
 #include <stdlib.h>
+#include <stdio.h>
```
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:2]: I like everything you done!

BUT!!!

File: ds/linked_list/src/ll.c [hunk 2]
```diff
@@ -1,14 +1,60 @@
...
+
+struct Node {
+    int data;
+    struct Node* next;
+};
+
+// Stack Implementation
+
+struct Node* top = NULL;
```
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:11]: Move into its own struct

File: ds/linked_list/src/ll.c [hunk 3]
```diff
@@ -1,14 +1,60 @@
...
+
+void push(int value){
```
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:13]: Spacing!  I don't have clang format set, but i'll get that shortly for this project.

Second, `return_type method_name(args) {` please :7

Also please name them with stack_push and stack_pop.  Just feels more ceezy :)

File: ds/linked_list/src/ll.c [hunk 4]
```diff
@@ -1,14 +1,60 @@
...
+    struct Node* new = (struct Node*)malloc(sizeof(struct Node));
+    new->data = value;
+    new->next = top;
+    top = new;
+    printf("\nYayayayayayayayaya pushed!!!\n\n");
```
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:18]: no prints!

Comment (ThePrimeagen) [reply ds/linked_list/src/ll.c:18]: There are a few of them, you keep them :)

File: ds/linked_list/src/ll.c [hunk 5]
```diff
@@ -1,14 +1,60 @@
...
+void peek(){
+    if(top == NULL){
+        return;
+    }
+
+    printf("%d\n", top->data);
+
+}
+
+void showAllElements(){
```
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:39]: this is excellent.  Could you make it work with both Q or S?

File: ds/linked_list/src/ll.c [hunk 6]
```diff
@@ -1,14 +1,60 @@
...
+void push(int value){
+    struct Node* new = (struct Node*)malloc(sizeof(struct Node));
+    new->data = value;
+    new->next = top;
+    top = new;
+    printf("\nYayayayayayayayaya pushed!!!\n\n");
+}
+
+
+void pop(){
```
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:22]: Please return the popp'd value.

File: ds/linked_list/src/ll.c [hunk 7]
```diff
@@ -1,14 +1,60 @@
...
+    if(top == NULL){
```
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:23]: also follow the style I was using.  I'll have to get clang format shortly.

File: ds/linked_list/src/ll.c (hunk 2 above)
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:11]: name top to tail.  that way it is consistent with the video :)

File: ds/linked_list/src/ll.c (hunk 5 above)
Comment (ThePrimeagen) [reply ds/linked_list/src/ll.c:39]: likely two methods, but you got it :)

File: ds/linked_list/src/ll.c [hunk 8]
```diff
@@ -53,5 +99,46 @@ int queue_pop(struct Queue* list) {
 }
 
 int main() {
+    // Testing the stack implementation
+
+    int value, given;
```
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:104]: Can we make it so you can create S or Q?

File: ds/linked_list/src/ll.c [hunk 9]
```diff
@@ -53,5 +99,46 @@ int queue_pop(struct Queue* list) {
...
+                showAllElements();
+
+            case 5:
+                break;
+            
+            default:
+                printf("Not a valid input!");
+        }
+    }
+    while(given!=5);
```
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:140]: style guide :)

**Review by ThePrimeagen — CHANGES_REQUESTED** [review]

**Comment by hemanth-kotagiri** [comment]
@ThePrimeagen just committed new changes. Check'em out! :smile:

File: ds/linked_list/src/ll.c (hunk 3 above)
Comment (hemanth-kotagiri) [reply ds/linked_list/src/ll.c]: Yes, done in the recent commit.

File: ds/linked_list/src/ll.c (hunk 4 above)
Comment (hemanth-kotagiri) [reply ds/linked_list/src/ll.c]: Okay.

File: ds/linked_list/src/ll.c (hunk 5 above)
Comment (hemanth-kotagiri) [reply ds/linked_list/src/ll.c]: Yes, made another function for Queue as well. Check it out!

File: ds/linked_list/src/ll.c (hunk 6 above)
Comment (hemanth-kotagiri) [reply ds/linked_list/src/ll.c]: Done!

File: ds/linked_list/src/ll.c (hunk 2 above)
Comment (hemanth-kotagiri) [reply ds/linked_list/src/ll.c]: Also done.

File: ds/linked_list/src/ll.c (hunk 8 above)
Comment (hemanth-kotagiri) [reply ds/linked_list/src/ll.c:154]: Yes, created two different prompt functions to create S or Q.

File: ds/linked_list/src/ll.c [hunk 10]
```diff
@@ -1,4 +1,89 @@
...
+#include <stdio.h>
+
+struct StackNode {
+    int data;
+    struct StackNode* next;
+};
+
+// Stack Implementation
+
+struct StackNode* tail = NULL;
```
Comment (ThePrimeagen) [inline ds/linked_list/src/ll.c:11]: Could you please put `tail` into a struct, like down below with `Queue`.  That way it feels ceezy.  Also shows that you can have many stacks easily.

Comment (ThePrimeagen) [reply ds/linked_list/src/ll.c:11]: You'll also notice that `Queue` keeps track of its size.  Please do that too.

```c
struct Queue {
    struct Node* head;	    struct Node* head;
    struct Node* tail;	    struct Node* tail;
    size_t size;	    size_t size;
};
```

**Review by ThePrimeagen — CHANGES_REQUESTED** [review]

File: ds/linked_list/src/ll.c (hunk 10 above)
Comment (hemanth-kotagiri) [reply ds/linked_list/src/ll.c:11]: Awesome. Will implement that. How is the style guide right now? is it okay?

Comment (ThePrimeagen) [reply ds/linked_list/src/ll.c:11]: Things are mostly there.

2 things:
1.  No omitted `{` from ifs (you have one with return)
2.  Don't treat if like a function.  i.e. `if(` but instead `if (`

**Review by ThePrimeagen — CHANGES_REQUESTED** [review]

**Pull request merged**